File: Frontend/src/events/converters/auctions/auction.converter.ts
import type { AuctionDTO } from "@/events/dto/auctions/auction.dto";
import type { Auction } from "@/models";

export class AuctionConverter {
    public static convert(auctionDto: AuctionDTO): Auction {
        return {
            id: auctionDto.id,
            hasParticipated: auctionDto.value !== undefined && auctionDto.value !== 0,
        };
    }
}

File: Backend/Events/Converters/AuctionConverter.cs
using Backend.Domain.Models;
using Backend.Events.DTO.Auctions;

namespace Backend.Events.Converters;

public static class AuctionConverter
{
    public static AuctionDTO ToDTO(Auction auction)
    {
        return new AuctionDTO
        {
            Id = auction.Id,
            Value = auction.Value,
        };
    }
}
